src/simulator_core/entities/esdl_object.py

```diff
@@ -1,18 +1,79 @@
+from simulator_core.entities.assets import AssetAbstract, DemandCluster, ProductionCluster, Pipe
 
+logger = logging.getLogger(__name__)
 
-class EsdlObject:
+
+class EsdlAssetObject:
     """
-    EsdlObject class is a wrapper around PyEsdl
+    Class to hold an esdl asset and convert it to local class objects.
+
+    Conversion is done based on the classes in the conversion_dict.
     """
-    es: esdl.EnergySystem
 
-    def __init__(self, esdl_energysystem: esdl.EnergySystem) -> None:
+    esdl_asset: esdl.Asset
+    conversion_dict = {
+        esdl.Producer: ProductionCluster,
+        esdl.GenericProducer: ProductionCluster,
+        esdl.Consumer: DemandCluster,
+        esdl.HeatingDemand: DemandCluster,
+        esdl.Pipe: Pipe
+    }
+
+    def __init__(self, asset: esdl.Asset) -> None:
+        """
+        Constructor for EsdlAssetObject class.
+
+        :param asset, esdl.Asset which os stored int hsi class for interaction.
+        """
+        self.esdl_asset = asset
+
+    def convert_esdl(self) -> AssetAbstract:
+        """Converts the esdl asset into the own class object."""
+        if not type(self.esdl_asset) in self.conversion_dict:
+            raise NotImplementedError(self.esdl_asset.__repr__() + ' not implemented in conversion')
+        return self.conversion_dict[type(self.esdl_asset)]()
+
+
+class EsdlObject:
+    """EsdlObject class is a wrapper around PyEsdl."""
+
+    esh: EnergySystemHandler
+
+    def __init__(self, esdl_energysystem_handler: EnergySystemHandler) -> None:
         """
-        constructor for EsdlObject
+        Constructor for EsdlObject.
+
         :param esdl_energysystem: PyEsdl EnergySystem object
         """
-        self.es = esdl_energysystem
+        self.esh = esdl_energysystem_handler
 
     def __repr__(self) -> str:
-        return str(self.es)
+        """Returns a string describing the esdl file object."""
+        return str(self.esh)
+
+    def get_all_assets_of_type(self, esdl_asset_type: str) -> list[EsdlAssetObject]:
+        """
+        Returns a list of all the esdl assets of the specified type in the esdl file.
+
+        If the type is not found an empty list is returned.
+        :param esdl_asset_type: str of the asset type assets need to be gathered.
+        """
+        str_to_type_dict = {
+            'asset': esdl.Asset,
+            'producer': esdl.Producer,
+            'consumer': esdl.Consumer,
+            'geothermal': esdl.GeothermalSource,
+            'conversion': esdl.Conversion,
+            'pipe': esdl.Pipe,
+            'transport': esdl.Transport,
+            'junction': esdl.Joint
+        }
+        if esdl_asset_type not in str_to_type_dict:
+            logger.error(esdl_asset_type + " not implemented in get_all_asset_of_type method")
+            return []
+        return [EsdlAssetObject(asset)
+                for asset in self.esh.get_all_instances_of_type(str_to_type_dict[esdl_asset_type])]
```

Comment:
Waarom dit?    Is het niet veel makkelijker om aparte functies te maken die de assets,. prodcuers, consumers, etc. terug geven?   Nu heb je 1 grote functie die zowel verstand moet hebben van PyEsdl terminologie als van onze interne entitiies?   (Single responsibility principle)

Reply:
Je moet eerst van de input string omzetten naar welke esdl assets je wilt hebben. Dat doe je moet de str_to_type_dict
Daarmee kun je dan lijst ophalen van esdl assets van dat type.
De esdlassetobject class die heeft de intiligentie om van esdl object naar onze eigen klassen te gaan, dit is apparte klasse die als verantwoordleijkheid heeft om esdl asset om te zetten naar onze eigen assets.
Dus het is zo wel al gescheiden

Reply:
@samvanderzwan de str_to_type_dict zou ook een externe "constante"/transformer kunnen zijn toch? Of zijn daar andere bezwaren voor?

Reply:
Dat zou kunnen, maar dan zet je die weer ergens anders en ben je soms lang aan het zoeken als je iets wilt toevoegen/wijzigen en nu staat het op de plek waar je het nodig hebt.

Reply:
@samvanderzwan   In dat geval is dit geen Entity class maar een mapper object en staat de file op de verkeerde plek (afgezien van de naam van de file, die niet klopt met de class naam).  Dan moet deze file dus verplaatst worden naar de adapter/transforms/mappers.  Daar staat nu een korte functie die voor elk asset object deze functie aanroept